lib/socket_server.rb

```diff
@@ -19,16 +19,22 @@ def stop
     @server.close if @server
     @server = nil
     @blk = nil
+    FileUtils.rm_f(@socket_path)
```

Comment:
need to protect against errors here, cause we have @server.close if @server ... similarly only rm if the socket path is there